src/pages/signin/SignInPageLayout/SignInPageContent.js

```diff
@@ -31,69 +29,60 @@ const propTypes = {
 };
 
 const SignInPageContent = props => (
-    <ScrollView
-        keyboardShouldPersistTaps="handled"
-        showsVerticalScrollIndicator={false}
+    <View
         style={[
-            styles.h100,
-            !props.isSmallScreenWidth && styles.alignSelfCenter,
-            !props.isSmallScreenWidth && styles.signInPageWideLeftContainer,
-        ]}
-        contentContainerStyle={[
-            scrollViewContentContainerStyles,
-            styles.alignItemsCenter,
-            !props.isSmallScreenWidth && styles.ph6,
+            styles.flex1,
+            styles.signInPageLeftContainer,
+
+            // Restrict the width if the left container only for large screens. For smaller screens, the width needs to be fluid to span the entire width of the page.
+            !props.isMediumScreenWidth && !props.isSmallScreenWidth && styles.signInPageWideLeftContainer,
         ]}
     >
-        <View style={[styles.flex1, styles.flexRow]}>
-            <View style={[
-                styles.flex1,
-                styles.signInPageNarrowContentContainer,
-            ]}
+        <ScrollView
+            style={[styles.h100]}
+            contentContainerStyle={[scrollViewContentContainerStyles]}
+            keyboardShouldPersistTaps="handled"
+            howsVerticalScrollIndicator={false}
+        >
+
+            {/* This empty view creates margin on the top of the sign in form which will shrink and grow depending on if the keyboard is open or not */}
```

Comment:
Hi @tgolen, thought it was important to note while investigating https://github.com/Expensify/App/issues/11277 I found removing this `ScrollView` fixes any of its `KeyboardAvoidingView` children that aren't functioning as expected.

I am a bit confused at to why we used this `ScrollView` in the first place; based on the expected behavior from https://github.com/Expensify/App/issues/11277 we never want this page to become scrollable, so why use a `ScrollView`?

Reply:
Sorry, got ahead of myself, just saw this: https://github.com/Expensify/App/pull/12848#issuecomment-1320503333, which explains why the `ScrollView` is necessary.

Reply:
Yeah, I definitely went back and forth about the scroll view. Maybe there is a better way to allow the keyboard to be dismissed than without a scroll view. Any ideas?

Reply:
We can use a `TouchableWithFeedback` and hook that up to `Keyboard.dismiss()`

Reply:
Like, it's weird that the input wouldn't just blur when you tap outside of it, right?

Reply:
https://reactnative.dev/docs/keyboard#dismiss

Reply:
I think the answer lies somewhere with this logic: I need to dig into it after lunch. https://github.com/Expensify/App/blob/0bbc47bbde1c2cda8ee4814ecd7f90c26d2f32f0/src/pages/signin/LoginForm.js#L91-L93

Reply:
```
function componentDidMount() {
    Keyboard.addListener('keyboardDidShow', () => this.setState({isKeyboardShowing: true}));
    Keyboard.addListener('keyboardDidHide', () => this.setState({isKeyboardShowing: false}));
}

function dismissKeyboardIfOpen() {
    if (!this.state.isKeyboardShowing) {
        return;
    }
    Keyboard.dismiss();
}
```

I'm going to try these out with what's still on `main` and but without the `ScrollView`.

If we decide to go this route it's probably cleanest to stick this in some HOC that has an index.ios.js file so we only use this logic for iOS.

Reply:
Why would we only want it for iOS? If it works for other platforms, then that's fine, isn't it?

Reply:
OK, Great! Then we just need to ensure this is the end result:

> Keyboard should stay active when transitioning to PasswordForm, but it should be dismissed when transitioning to ResendValidationForm

As well as allowing the inputs to be blurred when tapping outside of them (which should also hide the keyboard).

Reply:
https://github.com/Expensify/App/blob/0bbc47bbde1c2cda8ee4814ecd7f90c26d2f32f0/src/pages/signin/LoginForm.js#L91-L93

@tgolen I looked into the above logic wasn't able to see how we could use it to dismiss the keyboard. Removing it didn't allow tapping away from the keyboard, and I can't see how `componentDidUpdate` in that component would be invoked unless we somehow hooked up props/state somewhere to an `onPress` handler.

Reply:
Separately, I created a HOC `PressableDismissKeyboard` in this PR: https://github.com/Expensify/App/pull/12903:
- allows you to dismiss the keyboard on iOS. It basically does the hooking up of props/state to an `onPress` handler
- to use it you can change the `ScrollView` to a regular `View` and wrap the entire `SignInPageContent` component with it. Please let me know what you think about this

Reply:
Thanks, Jasper! I think that component idea is neat. I just can't help thinking though that it's covering up the root of the problem. There must be _something_ that's causing those inputs to not be able to lose focus. Any other normal form with text inputs works (by "works" I mean that tapping outside a text input dismisses the keyboard).

Let's keep digging!

Reply:
OK, I got this working, using your idea, Jasper. 

When I researched this, I found that I made a bad assumption. React-Native doesn't automatically dismiss the keyboard or blur an input when tapping outside of it. All the references I could find showed that it had to be manually done.

For now, I've only implemented this in the sign in form for now as I didn't want to cause a bunch of regressions in all the other forms.